src/main/java/io/Parser.java

```diff
@@ -0,0 +1,51 @@
+package io;
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.NoSuchElementException;
+/**
+ * Used parse the additional data
+ * given in a command based on tags.
+ * @author Alvis Ng (supermii2)
+ */
+public class Parser {
+    /** Untagged input*/
+    private String defaultInput;
+    /** Dictionary of tagged inputs*/
+    private Dictionary<String, String> taggedInputs;
+    /**
+     * Creates a parsed representation
+     * of the input string given by
+     * the user's input.
+     * @param input
+     */
+    public Parser(String input) {
+        taggedInputs = new Hashtable<>();
+        String[] phrases = input.split("/");
+        this.defaultInput = phrases[0];
+        for (int i = 1; i < phrases.length; i++) {
+            String[] words = phrases[i].split(" ", 2);
+            taggedInputs.put(words[0], words[1]);
+        }
+    }
+    /**
+     * Get the untagged input
+     * @return Untagged input of parsed message
+     */
+    public String getDefaultString() {
+        return defaultInput;
+    }
+    /**
+     * Get the tagged input with the
+     * given tag
+     * @param tag Tag used to find tagged input.
+     * @return Tagged input of the given tag
+     * @throws NoSuchElementException Tag not found in the inputs.
+     */
+    public String getTaggedInput(String tag) throws NoSuchElementException {
```

Comment:
I was not sure what "tag" refers to in this code and had to reread multiple times to find out. Consider giving variable names that correspond to what it refers to in real life, such as taskType, commandType or similar.

Reply:
In the example given, the input for, say, an event object is given by

```deadline [Name] /from [start] /to [end]```

The tag refers to the /from and /to added and is a way for commands to extract [start] by requesting the tag "from" from the parsed input.

Admittedly it isn't an intuitive name, but because this logic is shared by all commands (to prevent some boiler plate code), I couldn't come up with a less generic name unfortunately. However, I'll try to explain the tag system more clearly in the javadocs